### PR TITLE
Fixes #34713 - Remove pf3 OverlayTrigger

### PR DESCRIPTION
**Pull request by MariaAga**
tested each page (except the seperate datepicker and timepicker) and nothing broke.
had to add `.ellipsis-pf-tooltip` as some js was adding another tooltip to `.ellipsis`

### CODE REVIEW

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js [hunk 1]
```diff
@@ -1,22 +1,18 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import UUID from 'uuid/v1';
-import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react';
+import { Icon } from 'patternfly-react';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js:3]: While at it, can we get rid of `patternfly-react` Icon and use it from `'@patternfly/react-core'`?

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
 import { noop } from '../../../common/helpers';
 import { translate as __ } from '../../../common/I18n';
 
 const AutoCompleteClearButton = ({ onClear }) => (
-  <OverlayTrigger
-    overlay={<Tooltip id={UUID()}>{__('Clear')}</Tooltip>}
-    placement="top"
-    trigger={['hover', 'focus']}
-  >
+  <Tooltip content={__('Clear')} position={TooltipPosition.top}>
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js]: the default position is `'top'` so we don't have to specify it

File: webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js [hunk 3]
```diff
@@ -2,14 +2,9 @@
...
 import PropTypes from 'prop-types';
 
-import {
-  Button,
-  Icon,
-  OverlayTrigger,
-  FormControl,
-  Tooltip,
-} from 'patternfly-react';
+import { Button, Icon, FormControl } from 'patternfly-react';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:5]: can you transform `Button` and `Icon` also to use `'@patternfly/react-core'`?

File: webpack/assets/javascripts/react_app/components/Editor/components/EditorSettings.js [hunk 4]
```diff
@@ -1,13 +1,7 @@
...
 import PropTypes from 'prop-types';
-import {
-  Popover,
-  Dropdown,
-  MenuItem,
-  Button,
-  Icon,
-  OverlayTrigger,
-} from 'patternfly-react';
+import { Dropdown, MenuItem, Button, Icon } from 'patternfly-react';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Editor/components/EditorSettings.js:3]: if you can, here as well

File: webpack/assets/javascripts/react_app/components/Layout/components/Toolbar/InstanceTitleViewer.js [hunk 5]
```diff
@@ -1,22 +1,21 @@
 import React from 'react';
-import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react';
+import { Icon } from 'patternfly-react';
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Layout/components/Toolbar/InstanceTitleViewer.js:2]: can you switch it to PF4?

File: webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTableHelpers.js [hunk 6]
```diff
@@ -10,17 +9,13 @@ export const withTooltip = Component => componentProps => {
...
-    <OverlayTrigger
-      overlay={<Tooltip id={tooltipId}>{tooltipText}</Tooltip>}
-      trigger={['hover', 'focus']}
-      placement="top"
-      rootClose={false}
-    >
-      {/* The span is needed because OverlayTrigger overrides child events */}
+    <Tooltip content={tooltipText}>
+      {/* The span is needed because Tooltip overrides child events */}
+      {/* TODO check  */}
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTableHelpers.js]: leftover?

File: webpack/assets/javascripts/react_app/components/Editor/components/EditorSettings.js [hunk 7]
```diff
@@ -116,7 +115,7 @@ const EditorSettings = ({
     <Button className="editor-button" id="cog-btn" bsStyle="link">
       <Icon size="lg" name="cog" />
     </Button>
-  </OverlayTrigger>
+  </Popover>
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/Editor/components/EditorSettings.js:115]: there are a few leftovers from the old implementation like `placement`, `trigger`

File: webpack/assets/javascripts/react_app/components/common/ClipboardCopy/index.js [hunk 8]
```diff
@@ -21,11 +21,10 @@ const ClipboardCopy = ({
...
         {...textareaProps}
       />
-      <OverlayTrigger
-        overlay={<Tooltip id={UUID()}>{successMessage}</Tooltip>}
-        placement="right"
-        trigger={['click']}
-        rootClose
+      <Tooltip
+        content={successMessage}
+        placement={TooltipPosition.right}
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/common/ClipboardCopy/index.js]: `placement` isn't a Tooltip prop, can you rename it?

File: webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js [hunk 9]
```diff
@@ -103,7 +90,7 @@ DatePicker.propTypes = {
   locale: PropTypes.string,
   weekStartsOn: PropTypes.number,
   id: PropTypes.string,
-  placement: OverlayTrigger.propTypes.placement,
+  placement: PropTypes.string,
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js:93]: can you rename it to `position`?

File: webpack/assets/javascripts/react_app/components/common/DateTimePicker/DateTimePicker.js [hunk 10]
```diff
@@ -127,7 +116,7 @@ DateTimePicker.propTypes = {
   weekStartsOn: PropTypes.number,
   inputProps: PropTypes.object,
   id: PropTypes.string,
-  placement: OverlayTrigger.propTypes.placement,
+  placement: PropTypes.string,
```
Comment (Ron-Lavi) [inline webpack/assets/javascripts/react_app/components/common/DateTimePicker/DateTimePicker.js:119]: ditto

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js (hunk 1 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js:3]: out of the scope of this pr which is removing the overlaytrigger

File: webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js (hunk 3 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/Editor/components/EditorOptions.js:5]: out of the scope of this pr which is removing the overlaytrigger

File: webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTableHelpers.js (hunk 6 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTableHelpers.js]: yup, checked the comment and the `span` is still needed

File: webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js (hunk 9 above)
Comment (MariaAga) [reply webpack/assets/javascripts/react_app/components/common/DateTimePicker/DatePicker.js:93]: That might be a breaking change so I'll leave it

**Comment by MariaAga** [comment]
Thanks @Ron-Lavi for finding the small errors :) All should be gone now

File: webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/components/AutoComplete/components/AutoCompleteClearButton.js:3]: ok, though it's like a one-liner 👼🏼

**Review by Ron-Lavi — APPROVED** [review]
Thanks @MariaAga! great addition

**Pull request merged**